codec-http/src/main/java/io/netty/handler/codec/http/multipart/HttpPostMultipartRequestDecoder.java

```diff
@@ -805,6 +796,35 @@ private InterfaceHttpData findMultipartDisposition() {
         }
     }
 
+    private static Pattern doubleSingleQuote = Pattern.compile("''");
+
+    private Attribute getContentDispositionAttribute(String... values) {
+        String name = cleanString(values[0]);
+        String value = values[1];
+
+        boolean shouldDecode = false;
+        int last = name.length() - 1;
+        if (name.charAt(last) == '*' && HttpHeaderValues.FILENAME.contentEquals(name.substring(0, last))) {
```

Comment:
Would be better to reduce the possible allocations using `filename*` constant. Something like this:
```java
    private static final String FILENAME_ENCODED = HttpHeaderValues.FILENAME.toString() + '*';

    private Attribute getContentDispositionAttribute(String... values) {
        String name = cleanString(values[0]);
        String value = values[1];

        // See http://www.w3.org/Protocols/rfc2616/rfc2616-sec19.html
        if (HttpHeaderValues.FILENAME.contentEquals(name)) {
            // filename value is quoted string so strip them
            value = value.substring(1, value.length() - 1);
        } else if (FILENAME_ENCODED.equals(name)) {
            // filename value is encoded. See https://tools.ietf.org/html/rfc5987
            name = name.substring(0, name.length() - 1);
            String[] split = doubleSingleQuote.split(value, 2);
            value = QueryStringDecoder.decodeComponent(split[1], Charset.forName(split[0]));
        } else {
            // otherwise we need to clean the value
            value = cleanString(value);
        }
        return factory.createAttribute(request, name, value);
    }
```

Reply:
Thanks. Pushed this version.